pkg/billing/client.go

```diff
@@ -8,27 +8,40 @@ import (
 	"math/rand"
 	"net/http"
 	"os"
+	"sync/atomic"
 	"time"
 
 	"github.com/lithammer/shortuuid"
 )
 
 type Client struct {
-	URL      string
-	httpc    *http.Client
-	hostname string
+	URL   string
+	httpc *http.Client
 }
 
-func NewClient(url string, c *http.Client) Client {
-	hostname, err := os.Hostname()
+var hostname atomic.Pointer[string]
+
+func GetHostname() string {
```

Comment:
You can also do `init()` function, there will be no need for an atomic

https://www.digitalocean.com/community/tutorials/understanding-init-in-go

Reply:
Ohhh right, yeah fair enough. _Generally_ I seem to remember init is frowned upon, but maybe it's ok here :smile:

Reply:
Switched to init(). I did a brief search and it _does_ seem to be widely frowned upon, but I think this one is simple enough that we can get away with it.

Reply:
If you plan to rebase, you need to squash that commit elsewhere

Reply:
Oh right, good catch! thanks!